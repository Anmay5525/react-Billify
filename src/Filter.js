export default function Filter({ handleChange }) {
  return (
    <div>
      <input
        autoFocus
        className="filter-input"
        type="text"
        onChange={handleChange}
      ></input>
    </div>
  );
}
