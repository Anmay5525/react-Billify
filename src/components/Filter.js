import React from "react";
import PropTypes from "prop-types";

export default function Filter({ handleChange }) {
  return (
    <div>
      <input
        placeholder="Filter by name"
        className="filter-input"
        type="text"
        onChange={handleChange}
      />
    </div>
  );
}

Filter.propTypes = {
  handleChange: PropTypes.func.isRequired,
};
