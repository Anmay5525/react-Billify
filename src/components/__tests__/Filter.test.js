import React from 'react';
import ReactDOM, { render } from 'react-dom';
import renderer from 'react-test-renderer';
import Filter from '../Filter';

describe('Testing Filter component if it', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Filter handleChange={() => {}}></Filter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<Filter handleChange={() => {}}></Filter>).toJSON();
    expect(tree).toMatchSnapshot();
  });

});