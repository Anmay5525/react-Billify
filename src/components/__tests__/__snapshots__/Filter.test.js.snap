// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Filter component if it matches snapshot 1`] = `
"<Filter handleChange={[Function: handleChange]}>
  <div>
    <input placeholder=\\"Filter by name\\" className=\\"filter-input\\" type=\\"text\\" onChange={[Function: handleChange]} />
  </div>
</Filter>"
`;
